heim-virt/src/sys/linux/containers.rs

```diff
@@ -109,11 +109,11 @@ where
 pub async fn detect_container() -> Result<Virtualization, ()> {
     future::err(())
         .or_else(|_| detect_openvz())
-        .or_else(|_| detect_wsl("/proc/sys/kernel/osrelease"))
+        .or_else(|_| detect_wsl(format!("{}/sys/kernel/osrelease", rt::linux::procfs_root())))
         .or_else(|_| detect_systemd_container("/run/systemd/container"))
-        .or_else(|_| detect_init_env("/proc/1/environ"))
+        .or_else(|_| detect_init_env(format!("{}/1/environ", rt::linux::procfs_root())))
```

Comment:
```suggestion
        .or_else(|_| detect_init_env("/proc/1/environ"))
```

I'm pretty sure this should not be affected by the procfs override.